adr/ADR-8.md

```diff
@@ -360,15 +374,89 @@ first time any message has a `Pending==0`. This signal must also be sent if no d
 Whatchers should support at least the following options. Languages can choose to support more models if they wish, as long as that
 is clearly indicated as a language specific extension. Names should be language idiomatic but close to these below.
 
-|Name|Description|
-|----|-----------|
-|`IncludeHistory`|Send all available history rather than just the latest entries|
-|`IgnoreDeletes`|Only sends `PUT` operation entries|
-|`MetaOnly`|Does not send any values, only metadata about those values|
-|`UpdatesOnly`|Only sends new updates made, no current or historical values are sent. The End Of Initial Data marker is sent as soon as the watch starts.|
+| Name             | Description                                                                                                                                |
+|------------------|--------------------------------------------------------------------------------------------------------------------------------------------|
+| `IncludeHistory` | Send all available history rather than just the latest entries                                                                             |
+| `IgnoreDeletes`  | Only sends `PUT` operation entries                                                                                                         |
+| `MetaOnly`       | Does not send any values, only metadata about those values                                                                                 |
+| `UpdatesOnly`    | Only sends new updates made, no current or historical values are sent. The End Of Initial Data marker is sent as soon as the watch starts. |
 
 The default behavior with no options set is to send all the `last_per_subject` values, including delete/purge operations.
 
+#### Replicas for NATS Server 2.10 and newer
+
+Since NATS Server 2.10 we support transforming messages as a stream configuration item. This allows us to source one 
+bucket from another and rewrite the keys in the new bucket to have the correct name.
+
+To copy the keys `NEW.>` from bucket `ORDERS` into `NEW_ORDERS` we create the new stream with the following 
+partial config:
+
+```json
+    "sources": [
+      {
+        "name": "KV_ORDERS",
+        "filter_subject": "$KV.ORDERS.NEW.>",
+        "subject_transforms": [
+          {
+            "src": "$KV.ORDERS.>",
+            "dest": "$KV.NEW_ORDERS.>"
+          }
+        ]
+      }
+    ],
+```
+
+This results in all messages from `ORDERS` keys `NEW.>` to be copied into `NEW_ORDERS` and the subjects rewritten on 
+write to the new bucket so that a unmodified KV client on `NEW_ORDERS` would just work.
+
+As this is a `Source` and not a `Mirror` this new bucket can accept writes.  Sourced streams can be created with no
+listening subjects which would render the mirror a read-only replica.  Given that it's a Source it can be made with
+multiple sources to create an aggregate bucket by using multiple subject transforms.
```

Comment:
@jnmoyne we should think about this, today in nats.go for example we always create a listening subject and have no way to opt out of that for a sourced stream, its quite dangerous.

Reply:
We should also allow things like read-only sourced replicas which would have allow delete, allow purge, allow rollup all as false.

We should never have used the standard SourceInfo structure to configure this stuff in KV.

Reply:
As discussed on zoom, we'll add new configs for the streams that are replicas rather than the main origin bucket.  We'll deprecate entirely the mirror/source config in KeyValueConfig.  This way we manage topology different from origin which feels safer and more maintainable.